addons/l10n_it_edi/models/account_move.py

```diff
@@ -0,0 +1,1212 @@
+
+            description = line.name
+
+            # Down payment lines:
+            # If there was a down paid amount that has been deducted from this move,
+            # we need to put a reference to the down payment invoice in the DatiFattureCollegate tag
+            downpayment_moves = self.env['account.move']
+            if not is_downpayment and line.price_subtotal < 0:
+                downpayment_moves = line._get_downpayment_lines().mapped("move_id")
+                if downpayment_moves:
+                    downpayment_moves_description = ', '.join(m.name for m in downpayment_moves)
+                    sep = ', ' if description else ''
+                    description = f"{description}{sep}{downpayment_moves_description}"
+
+            invoice_lines.append({
+                'line': line,
+                'line_number': num + 1,
+                'description': description or 'NO NAME',
+                'unit_price': price_unit,
+                'subtotal_price': price_subtotal,
+                'vat_tax': line.tax_ids._l10n_it_filter_kind('vat'),
+                'downpayment_moves': downpayment_moves,
+                'discount_type': (
+                    'SC' if line.discount > 0
+                    else 'MG' if line.discount < 0
+                    else False)
+            })
+        return invoice_lines
+
+    def _l10n_it_edi_get_tax_values(self, tax_details):
+        """ Returns a list of dictionaries passed to the template for the invoice lines (DatiRiepilogo)
+        """
+        tax_lines = []
+        for _tax_name, tax_dict in tax_details['tax_details'].items():
+            # The assumption is that the company currency is EUR.
+            base_amount = tax_dict['base_amount']
+            tax_amount = tax_dict['tax_amount']
+            tax_rate = tax_dict['tax'].amount
+            expected_base_amount = tax_amount * 100 / tax_rate if tax_rate else False
+            tax = tax_dict['tax']
+            # Constraints within the edi make local rounding on price included taxes a problem.
+            # To solve this there is a <Arrotondamento> or 'rounding' field, such that:
+            #   taxable base = sum(taxable base for each unit) + Arrotondamento
+            if tax.price_include and tax.amount_type == 'percent':
+                if expected_base_amount and float_compare(base_amount, expected_base_amount, 2):
+                    tax_dict['rounding'] = base_amount - (tax_amount * 100 / tax_rate)
+                    tax_dict['base_amount'] = base_amount - tax_dict['rounding']
+
+            tax_line_dict = {
+                'tax': tax,
+                'rounding': tax_dict.get('rounding', False),
+                'base_amount': tax_dict['base_amount'],
+                'tax_amount': tax_dict['tax_amount'],
+            }
+            tax_lines.append(tax_line_dict)
+        return tax_lines
+
+    def _l10n_it_edi_filter_tax_details(self, line, tax_values):
```

Comment:
`@api.model`

Reply:
I don't see why this is needed since this will never be called via json-rpc